### PR TITLE
Feat: Topic page with list

### PR DESCRIPTION
**Pull request by jimhunty**
This PR features:
* Article List in the Topic Page. This has been modified to closely resemble Author Profile
* Updated Topic Provider again to closer resemble Author Profile
* Small update to Article List to support Byline
* Fixture Generator is a quick copy/paste job. Refactor is being handled by @colincclark so ignore for now.
* Previous Topic file tests have been moved to TopicHead

### CODE REVIEW

**Comment by times-tools** [comment]
If you use Expo, view our components by scanning this qr code: <br>  <br> This has been made possible through [Fructose](https://github.com/newsuk/fructose)

File: packages/topic/__tests__/web/responsive.web.test.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ import {
   ResponsiveDivider
 } from "../../src/styles/responsive.web";
 
-describe("Topic responsive tests on web", () => {
+describe("TopicHead responsive tests on web", () => {
```
Comment (colincclark) [inline packages/topic/__tests__/web/responsive.web.test.js]: This will test all `Topic` responsive components

File: packages/topic/package.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
     "test:ios": "jest --config='./__tests__/ios/jest.config.js'",
     "test:web": "jest --config='./__tests__/web/jest.config.js'",
     "test:all": "yarn test:android --coverage & yarn test:ios --coverage & yarn test:web --coverage",
-    "prepublishOnly": "npm run transpile",
```
Comment (colincclark) [inline packages/topic/package.json]: github issue to add this to the cli?

File: packages/topic/src/topic-head-content.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import React, { Fragment } from "react";
 import { Text, View } from "react-native";
-import { propTypes, defaultProps } from "./proptypes";
+import { propTypes, defaultProps } from "./topic-head-proptypes";
```
Comment (colincclark) [inline packages/topic/src/topic-head-content.js]: file name should be `topic-head-prop-types` please

File: packages/topic/src/topic-head-proptypes.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+export const topicHeadPropTypes = {
+  name: PropTypes.string,
+  description: PropTypes.string,
+  isLoading: PropTypes.bool
+};
+
+export const topicHeadDefaultProps = {
+  name: "Animals",
+  description: "Animals are multicellular eukaryotic organisms.",
+  isLoading: false
```
Comment (colincclark) [inline packages/topic/src/topic-head-proptypes.js]: `isLoading` states should be true by default I think

File: packages/topic/topic.showcase.js [hunk 5]
```diff
@@ -1,41 +1,53 @@
...
 
+const preventDefaultedAction = decorateAction =>
+  decorateAction([
+    ([e, ...args]) => {
+      e.preventDefault();
+      return ["[SyntheticEvent (storybook prevented default)]", ...args];
+    }
+  ]);
+
+const page = 2;
```
Comment (colincclark) [inline packages/topic/topic.showcase.js]: should start at page 1

File: packages/topic/__tests__/ios/topic-functional.test.js [hunk 6]
```diff
@@ -1,9 +1,5 @@
-import { mockReactNativeComponent } from "@times-components/jest-configurator";
-import sharedFunctional from "../shared-functional";
+import sharedStyling from "../topic-functional";
```
Comment (aronshamash) [inline packages/topic/__tests__/ios/topic-functional.test.js]: update name

File: packages/topic/topic.showcase.js [hunk 7]
```diff
@@ -1,41 +1,53 @@
...
+          analyticsStream: storybookReporter,
+          onArticlePress: preventDefaultedAction(decorateAction)(
+            "onArticlePress"
+          ),
+          slug,
+          page,
+          pageSize,
+          topic: {
+            name: "Chelsea",
+            description: "A swanky part of town."
```
Comment (colincclark) [inline packages/topic/topic.showcase.js]: worth making these knobs, or keep the knobs in article-list only? up to you...

File: packages/topic/src/topic-head-proptypes.js (hunk 4 above)
Comment (aronshamash) [reply packages/topic/src/topic-head-proptypes.js]: author-profile has it false by default

**Comment by colincclark** [comment]
Could the storybook benefit from a story showing no description? Also, one with an error, etc? Similar to author-profile...

File: packages/topic/topic.showcase.js (hunk 7 above)
Comment (aronshamash) [reply packages/topic/topic.showcase.js]: I think these can be knobs here. You mean for the optional description ?

Comment (colincclark) [reply packages/topic/topic.showcase.js]: yes

File: packages/topic/src/topic-head-proptypes.js (hunk 4 above)
Comment (colincclark) [reply packages/topic/src/topic-head-proptypes.js]: i changed it in the refactor

File: packages/provider-test-tools/fixtures/topic.json [hunk 8]
```diff
@@ -1,29 +1,40 @@
...
+            "label": "Middle East",
             "leadAsset": {
-              "id": "22b95cda-d725-4467-c8f6-b24024369af8",
-              "title": "This photo taken in Kenya and ha",
+              "id": "86ee3055-9780-4732-a210-1e3987d84099",
+              "title": "Splits in Europe may end Russian sanctions",
               "crop": {
                 "url":
-                  "//nu-cps-imgsrv-tnl-dev-webapp.elb.tnl-dev.ntch.co.uk/imageserver/image/26402522-4a2d-11e8-8db5-58268675bbb1?crop=2787%2C1858%2C634%2C445",
+                  "//nu-cps-imgsrv-tnl-dev-webapp.elb.tnl-dev.ntch.co.uk/imageserver/image/e21fcb53756c321b9894c0650968fb9bc74197fb.jpg?crop=780%2C520%2C0%2C0",
```
Comment (colincclark) [inline packages/provider-test-tools/fixtures/topic.json]: these do not show in native

File: packages/topic/__tests__/topic-head-functional.js [hunk 9]
```diff
@@ -27,7 +27,9 @@ export default () => {
   });
 
   it("should render correctly when loading", () => {
-    const tree = renderer.create(<Topic name="Animals" isLoading />).toJSON();
+    const tree = renderer
+      .create(<TopicHead name="Animals" isLoading />)
```
Comment (colincclark) [inline packages/topic/__tests__/topic-head-functional.js]: the loading component is not covered in coveralls for some reason, but the divider is.... weird

File: packages/topic/src/topic-head-proptypes.js (hunk 4 above)
Comment (aronshamash) [reply packages/topic/src/topic-head-proptypes.js]: ok will do accordingly here then

File: packages/topic/topic.showcase.js (hunk 7 above)
Comment (colincclark) [reply packages/topic/topic.showcase.js]: I did this here: https://github.com/newsuk/times-components/pull/950/files#diff-57f0475f2fb616b6b682d04bd8bd9dbaR7 (still to come)

Comment (colincclark) [reply packages/topic/topic.showcase.js]: so don't worry about it

File: packages/provider-test-tools/fixtures/topic.json (hunk 8 above)
Comment (aronshamash) [reply packages/provider-test-tools/fixtures/topic.json]: yes I think they need to be on `www.` as that domain is not secure

File: packages/topic/__tests__/topic-head-functional.js (hunk 9 above)
Comment (jimhunty) [reply packages/topic/__tests__/topic-head-functional.js]: Once we do the next part of the work, this should definitely get covered

File: packages/topic/package.json (hunk 2 above)
Comment (jimhunty) [reply packages/topic/package.json]: Go right ahead 😉

Comment (colincclark) [reply packages/topic/package.json]: :trollface:

**Pull request merged**